clang/lib/DPCT/APIMapping/QueryAPIMapping.cpp

```diff
@@ -27,34 +27,33 @@ void APIMapping::registerEntry(std::string Name, llvm::StringRef SourceCode) {
   const auto TargetIndex = EntryArray.size();
   EntryMap[Name] = TargetIndex; // Set the entry whether it exist or not.
   // Try to fuzz the original API name (only when the entry not exist):
-  // 1. Remove partial or all leading '_'.
-  // 2. For each name got by step 1, put 4 kind of fuzzed name into the map
+  // 1. Change "Name" to lower case. (Querying will change "Key" to lower too)
+  // 2. Remove partial or all suffix '_'.
+  std::transform(Name.begin(), Name.end(), Name.begin(), ::tolower);
+  while (Name.back() == '_') {
+    Name.erase(Name.end() - 1);
+    EntryMap.try_emplace(Name, TargetIndex);
+  }
+  const auto EmplaceWithAndWithoutSuffix =
+      [TargetIndex](std::string Name, const std::string Suffix) {
```

Comment:
```suggestion
      [TargetIndex](StringRef Name,StringRef Suffix) {
```

Reply:
only can change the second one to `StringRef`.

Reply:
Maybe it would be better to use llvm::Twine::concat to concat string.

Reply:
fixed